frida_tools/repl.py

```diff
@@ -757,6 +760,92 @@ def _cancel_line(self):
+                    return self.complete(*cargs, **ckargs)
+                self.register_function(complete)
+
+            def print_port(self, stream=None):
+                if stream is None:
+                    stream = sys.stdout
+                stream.write('___EPCCompletionServer_PORT=')
+                EPCServer.print_port(self, stream)
+
+        class EPCCompletionClient(EPCClient):
+            def __init__(self, address='localhost', port=None, *args, **kargs):
+                if port is not None:
+                    args = ((address, port), ) + args
+                EPCClient.__init__(self, *args, **kargs)
+
+                def complete(*cargs, **ckargs):
+                    return self.complete(*cargs, **ckargs)
+                self.register_function(complete)
+
+        EpcDocument = namedtuple('Document', ['text_before_cursor', ])
+
+        class ReplEPCCompletion(object):
+            def __init__(self, repl, *args, **kargs):
+                self._repl = repl
+
+            def complete(self, *to_complete):
+                to_complete = ''.join(to_complete)
+                prefix = '.'.join(to_complete.split(".")[:-1])
+                if len(prefix) > 0:
+                    prefix += "."
+                completions = [
+                    prefix + c.text
+                    for c in self._repl._completer.get_completions(
+                        EpcDocument(text_before_cursor=to_complete), None)
+                ]
+                return tuple(completions)
+
+        class ReplEPCCompletionServer(EPCCompletionServer, ReplEPCCompletion):
+            def __init__(self, repl, *args, **kargs):
+                EPCCompletionServer.__init__(self, *args, **kargs)
+                ReplEPCCompletion.__init__(self, repl)
+
+        class ReplEPCCompletionClient(EPCCompletionClient, ReplEPCCompletion):
+            def __init__(self, repl, *args, **kargs):
+                EPCCompletionClient.__init__(self, *args, **kargs)
+                ReplEPCCompletion.__init__(self, repl)
+
+        def start_completion_thread(repl, epc_port=None):
+            if epc_port is None:
+                epc_port = os.environ.get("EPC_COMPLETION_SERVER_PORT", None)
+            if epc_port is not None:
+                epc_port = int(epc_port)
+                rpc_complete = ReplEPCCompletionClient(repl, port=epc_port)
+                rpc_complete_thread = threading.Thread(
+                    target=rpc_complete.connect,
+                    name='PythonModeEPCCompletion',
+                    kwargs={'socket_or_address': ('localhost', epc_port)})
+            else:
+                rpc_complete = ReplEPCCompletionServer(repl)
+                rpc_complete.print_port()  # needed for Emacs client
+                rpc_complete_thread = threading.Thread(
+                    target=rpc_complete.serve_forever,
+                    name='PythonModeEPCCompletion')
+            rpc_complete_thread.setDaemon(True)
```

Comment:
Let's use the modern API instead:

```suggestion
            rpc_complete_thread.daemon = True
```

That way we also avoid the camelCase method breaking the style. :-)